### DISCUSSION TITLE
Add Support To Enchant Modded Items

### DISCUSSION THREAD

**Discussion by RupticMC** [Zrips/CMI#3877] [Feature Requests]
**Description of feature/improvement request:**
Add the ability to enchant modded items (only tested on 1.7.10, have not tried for 1.12 but I assume 1.12 has even less support for modded) like essentials is able to.
---
### Detailed explanation of what it should do, when and in what situation: ###
When doing /enchant with a modded item in hand, you should be able to add any enchantment you want.

### Basic explanation as to why this feature is needed and/or useful to have: ###
Useful to put those enchantments on modded items like a pickaxe from Biomes O Plenty.

Gives an error when trying... https://paste.helpch.at/ojujecexud.apache

**Comment by CloudeLecaw** [comment]
you forgot to inform us about the server and CMI-Version your using,.....

**Comment by RupticMC** [comment]
> you forgot to inform us about the server and CMI-Version your using,.....

Cmi Version (using/cmi version):
8.7.5.0
Server Type (Spigot/Paperspigot/etc):
UraniumMC (Basically a better version of Thermos/Cauldron for modded 1.7.10) Uses Spigot...
Uranium version 1710-dev-5-B271-31d6587
Server Version (using /ver):
dev 1.7.10-R0.1-SNAPSHOT

**Comment by CloudeLecaw** [comment]
sorry.
CMI is not tested anymore for 1.7.10 - so features can run, but dont need to.
> Tested Minecraft Versions:
> 1.8 1.9 1.10 1.11 1.12 1.13 1.14 1.15 1.16

The requirement of the Wiki are not really up to date anymore. You will see the CMi will still run on 1.7.10 but newer feature of it do not run, and you can apect errors when using them on 1.7.10 servers.

Here u have a little thread where someone hada also a problem with his 1.7.10 server and you also can view the reviews of zrips and some users there

https://github.com/Zrips/CMI/issues/2549

Personally i would rate you to upgrade your server

![image](https://user-images.githubusercontent.com/12080017/89168029-80b18500-d57c-11ea-9fcb-0fd7b05d7cb2.png)
you see it over 6 years now since spigot was made for 1.7.10 - i badly recomment you to upgrade now to an other version

About your feature i see no problem in 1.16.1 with enchanting moded items as long the mod enchants dont use the same ntb as the enchant^^(should be fixed by the moded item plugin/mod)


2. point: you used a not registered enchant

> INFO AllAboutRuptic issued server command: /cmi enchant dig_speed 1

There is also no enchant like dig_speed ;)
You have efficiency - for tools
And you have haste as effect

so back to the table..
https://minecraft.gamepedia.com/Enchanting  and https://minecraft.gamepedia.com/Status_effect

dig_speed :3...... may you should thought about be able to add timed effects (or endless with level) with enchant command?

This would be a feature - but dig speed is no enchant at the moment so /cmi enchant does not work until you translated dig_speed for an enchant like effency in the langfile ;) then you could use /cmi enchant dig_speed


So what final to say for today?
-------------------------------------
> /cmi enchant walk_speed 5

;-)
